packages/frontend/@n8n/design-system/src/v2/components/DropdownMenu/DropdownMenuSearch.vue

```diff
@@ -0,0 +1,126 @@
+<script setup lang="ts">
+import { ref, useCssModule } from 'vue';
+
+import Icon from '@n8n/design-system/components/N8nIcon/Icon.vue';
+
+defineOptions({ name: 'N8nDropdownMenuSearch' });
+
+withDefaults(
+	defineProps<{
+		modelValue?: string;
+		placeholder?: string;
+		showIcon?: boolean;
+	}>(),
+	{
+		modelValue: '',
+		placeholder: 'Search...',
+		showIcon: true,
+	},
+);
+
+const emit = defineEmits<{
+	'update:modelValue': [value: string];
+	'key:escape': [];
+	'key:navigate': [direction: 'up' | 'down'];
+	'key:arrow-right': [];
+	'key:arrow-left': [];
+	'key:enter': [];
+}>();
+
+const $style = useCssModule();
+const inputRef = ref<HTMLInputElement | null>(null);
+
+const handleInput = (event: Event) => {
+	if (event.target instanceof HTMLInputElement) {
+		emit('update:modelValue', event.target.value);
+	}
+};
+
+const handleKeydown = (event: KeyboardEvent) => {
+	if (event.key === 'Escape') {
+		emit('key:escape');
+	} else if (event.key === 'Tab' && !event.shiftKey) {
+		event.preventDefault();
+		emit('key:navigate', 'down');
+	} else if (event.key === 'ArrowDown') {
+		event.preventDefault();
+		emit('key:navigate', 'down');
+	} else if (event.key === 'ArrowUp') {
+		event.preventDefault();
+		emit('key:navigate', 'up');
+	} else if (event.key === 'ArrowRight') {
+		if (
+			event.target instanceof HTMLInputElement &&
+			event.target.selectionStart === event.target.value.length
+		) {
+			emit('key:arrow-right');
+		}
+	} else if (event.key === 'ArrowLeft') {
+		if (event.target instanceof HTMLInputElement && event.target.selectionStart === 0) {
+			emit('key:arrow-left');
+		}
+	} else if (event.key === 'Enter') {
+		event.preventDefault();
+		emit('key:enter');
+	}
+};
+
+const focus = () => {
+	inputRef.value?.focus();
+};
+
+defineExpose({ focus, inputRef });
+</script>
+
+<template>
+	<div :class="$style['search-container']">
+		<Icon
+			v-if="showIcon"
+			:class="$style['search-icon']"
+			icon="search"
+			size="large"
+			color="text-light"
+		/>
```

Comment:
We don't ever want to allow an icon in this search field. Will look too similar to one of the menu items if we do so this can be removed.